rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -63,18 +75,21 @@ public void unregisterGlobalErrorCallback(OnErrorCallback globalError) {
    * Inner class responsible for routing the errors thrown from the use case to the error handler
    */
   private static class OnErrorCallbackToErrorHandlerAdapter implements OnErrorCallback {
-    private final ErrorHandler errorHandler;
-    private final OnErrorCallback useCaseOnErrorCallback;
+    private final WeakReference<ErrorHandler> errorHandler;
+    private final WeakReference<OnErrorCallback> useCaseOnErrorCallback;
 
     public OnErrorCallbackToErrorHandlerAdapter(ErrorHandler errorHandler,
         OnErrorCallback useCaseOnErrorCallback) {
-      this.errorHandler = errorHandler;
-      this.useCaseOnErrorCallback = useCaseOnErrorCallback;
+      this.errorHandler = new WeakReference<>(errorHandler);
+      this.useCaseOnErrorCallback = new WeakReference<>(useCaseOnErrorCallback);
     }
 
     @Override public boolean onError(Error error) {
-      errorHandler.notifyError(error, useCaseOnErrorCallback);
-      return true;
+      if (errorHandler.get() != null) {
+        errorHandler.get().notifyError(error, useCaseOnErrorCallback.get());
```

Comment:
Just to be sure we don't have any race condition related to the GC stage we should change this segment of code to check the ``errorHandler.get()`` nullity and use it over the same variable. 

```java
ErrorHandler errorHandler = errorHandler.get();
if (errorHandler != null) {
         errorHandler.notifyError(error, useCaseOnErrorCallback.get());
}
```